### ISSUE TITLE
Flag for enabling/disabling error handling

### ISSUE THREAD

**Issue by fabiantheblind** [basiljs/basil.js#35]
Hi guys.
To clean up that discussion I'll move the error handling mentioned in #33 discussion into an own issue. I will be working on this the next days.  

- [ ] add global boolean flag `pub.ERROR_HANDLING`
- [ ] disable enable error handling by basil
- [ ] run tests

Location of error handling --> https://github.com/basiljs/basil.js/blob/master/includes/core.js#L435

**Comment by trych** [comment]
Quick question about this:
What would be the advantage for the end user to disable error handling? Or is this intended for developers?

On the other hand: Now that I started working on the basil.js code, I could really use some nicer error handling (for developers), something that returns line number and file. Is there any plans to implement that?

(I guess, I could create my own error handling branch for debugging, but maybe others would find it useful, if that gets implemented in general).

**Comment by fabiantheblind** [comment]
I took a look at the error handling and wrote some code for it. Take a look here https://github.com/fabiantheblind/basil.js/tree/errorhandling   

Maybe I should pull this branch into the basil account. Makes no sense to have it on mine.

**Comment by fabiantheblind** [comment]
pulled it into a new branch here https://github.com/basiljs/basil.js/tree/errorhandling

**Comment by trych** [comment]
Unfortunately the branch suffers from the same issue as my closed PR earlier, there is so much deletion of trailing white space going on that it is really hard to review.

How about we do actually a single PR that removes trailing white spaces only (that will be a lot, but it will be easy to review, probably take only a few minutes), then merge that and then for every change after that it will be much easier to review. Before anything else, even before the implementation of proper linting. @fabiantheblind what do you think?

**Comment by fabiantheblind** [comment]
Yeah. There are some whitespaces removed but the diff of this file is not that complex https://github.com/basiljs/basil.js/commit/46c68958c6259b425e3378848c42d5043f19facc
